### PR TITLE
@swm/navigation refactor

### PR DESCRIPTION
**Pull request by WoLewicki**
### Details

Copy of https://github.com/Expensify/App/pull/18605 with the newest `main` branch merged.

### Fixed Issues

$ https://github.com/Expensify/App/issues/19665

### Tests

Most of the regression testing will come from Applause running through the App tests and logging any navigation errors (unusual results). Since there are virtually limitless combinations and flows within the App, we will only create a couple specific ones which we will ask Applause to go through and we can use to test the PRs. We would also ask QA to run full regression on the main PR before we merge these changes.

Note: For all of these, we should test them on all platforms (except resizing on mobile)

Deep link to authenticated page when signed out (pending route)

1. In Notes app for example, write a link to Settings page: https://new.expensify.com/settings
1. Be signed out on the testing device
1. Click on the link
1. Verify you have been navigated to the sign in flow
1. Sign in
1. Verify you have been correctly navigated to the Settings page after signing in

Deep link from notification

1. Have the App in the background on your testing device signed into User A account
1. From User B account, send a message to to User A
1. As user A, click on the new message notification
1. Verify the App has opened and you have been navigated to the correct chat

Correct Back button behavior when deep linking

1. Be signed in to the App on you testing device
1. In Notes app for example, write a link to Settings page: https://new.expensify.com/settings
1. Click on the link and verify you have been navigated to the Setting page
1. Click back button (either in desktop web or Android physical back button) 
1. Verify you have been navigated to the Notes app (or wherever you have clicked the link on)

Smooth navigation after resizing

1. On web, in wide layout, sign in and navigate to the Concierge chat
1. Resize the browser window so its narrow enough so it switches to the mobile layout
1. Verify that you are in the Concierge chat
1. Click the UP button
1. Verify you have been navigated to the LHN
1. Navigate to Settings page
1. Resize the window back to wide view
1. Verify the Settings page is open in the RHP
1. Verify you can see the Concierge chat in the background

Verify Up button behavior after navigating to a nested page

1. Send a message which includes this link to someone: https://new.expensify.com/settings/about
1. Click on the link from the message
1. Verify you have been navigated to the About page
1. Click Back button (either browser back button or Android physical back button)
1. Verify you have been navigated back to the chat
1. Click the link again to get to the About page
1. Click the Up button
1. Verify you have been navigated to the Settings page
1. Click the Up button again
1. Verify you have been navigated to the original chat

Test all deep links

Note: We had issues with deep links recently and to prevent this we will add new regression tests which will test all the possible deep link routes in the App. We won't write them down here as it's not tightly connected to the core idea of this doc, but noting it here so we don’t forget about this.

- [x] Verify that no errors appear in the JS console

### QA Steps

Same as tests

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/32481228/0076f24a-4693-4b6e-accd-47b113c332ac

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/32481228/bc118b39-dfd1-4025-9d80-8f78e069d552

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/32481228/ac4fcf4c-849e-4007-a61d-dfe4eb871229

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/32481228/a0f088b1-3e9a-4e3f-8f76-022b78d35a5a

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/32481228/6698ecbe-2a48-4563-b275-12c178679e83

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/32481228/99a008c2-5b44-4684-a5a8-2d23de99b231

</details>

### CODE REVIEW

**Comment by mountiny** [comment]
QA results so far:
- https://github.com/Expensify/App/issues/19384 I have seen this when using the app, but its not consistent, sometimes the composer is focused sometimes its not, not sure what are the best reproduction steps
- https://github.com/Expensify/App/issues/19383 this one might be from main, I dont think its related to our branch - we can build new version later and confirm
- https://github.com/Expensify/App/issues/19381 - this is definitely a bug
- https://github.com/Expensify/App/issues/19380 - I think this is related to the url matching logic and it should work in staging and production

**Review by mountiny — COMMENTED** [review]
@WoLewicki Couple of comments from the PR, what an amazing job SWM has done here, really!

I am using the internal release build and its really nice, excited to send this out once we polish the last couple of things.

File: ios/Podfile.lock
Comment (WoLewicki) [reply ios/Podfile.lock:1022]: They shouldn't be. I'll revert all changes from this file.

**Review by WoLewicki — COMMENTED** [review]
A quick self review with some comments.

As for working on navigation after the PR is merged, here is a couple important things:

- Since you can deeplink to different pages inside the RHP navigator, it is important to provide the possibility for the user to properly navigate back from any page with UP press (`HeaderWithBackButton` component). An example can be deeplinking to `/settings/profile/personal-details`. From there, when pressing the `UP` press, you should navigate to `/settings/profile`, so in order for it to work, you should provide the correct route in `onBackButtonPress` prop of `HeaderWithBackButton` (see https://github.com/adamgrzybowski/App/blob/66e959afff9d85d3c9cdf95ca048ad25137acdca/src/pages/settings/Profile/PersonalDetails/PersonalDetailsInitialPage.js#L94 for this example)
- New RHP flows should be added in `RigthModalNavigator`
- If you want to navigate to a certain report after completing a flow regarding it, e.g. `RequestMoney` flow with certain group/user, you should use `Navigation.dismissModal` with this reportID as an argument (see https://github.com/adamgrzybowski/App/blob/66e959afff9d85d3c9cdf95ca048ad25137acdca/src/libs/actions/IOU.js#L368 for this example). If in future we would like to navigate to something different than the report after such flows, the API should be rather easy to be changed.

**Comment by mountiny** [comment]
@WoLewicki i dont think the latest commit has been correct

**Comment by mountiny** [comment]
Not on Hold anymore, but we need to clean this up, the latest commit had incorrect merge I believe.

**Comment by mountiny** [comment]
This is not on hold anymore, but we should clean this up and make internal release again

File: src/pages/home/sidebar/SidebarScreen/index.js
Comment (mountiny) [inline src/pages/home/sidebar/SidebarScreen/index.js:30]: @adamgrzybowski @WoLewicki With moving the freeze up here, I think we should make sure the FAB button opens up fine when user signs in. ie when you create a new account and user loads the app first, it should show the menu with Request money etc open

Comment (mountiny) [reply src/pages/home/sidebar/SidebarScreen/index.js:30]: image

worked :)

File: src/libs/Navigation/FreezeWrapper.js
Comment (mountiny) [inline src/libs/Navigation/FreezeWrapper.js:26]: what is `o` here, lets try to be more descriptive

**Comment by mountiny** [comment]
Triggered a build here https://github.com/Expensify/App/actions/runs/5191341385

File: tests/ui/UnreadIndicatorsTest.js
Comment (mountiny) [inline tests/ui/UnreadIndicatorsTest.js:460]: I can confirm this works fine in the app so we can fix these tests later.

File: tests/ui/UnreadIndicatorsTest.js
Comment (mountiny) [inline tests/ui/UnreadIndicatorsTest.js:378]: This also works fine, I would create a follow up issues to re-enable these tests and fix whatever is causing them to fail

**Comment by mountiny** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Comment by mountiny** [comment]
@0xmiroslav Has reviewed most of this code already as part of the other PRs.

**Comment by WoLewicki** [comment]
@0xmiroslav done 🚀

File: src/components/withCurrentReportId.js
Comment (mountiny) [inline src/components/withCurrentReportId.js:14]: We need to make this functional component

File: src/libs/Navigation/AppNavigator/Navigators/CentralPaneNavigator.js
Comment (mountiny) [inline src/libs/Navigation/AppNavigator/Navigators/CentralPaneNavigator.js:20]:
```suggestion
                    // We do it this way to avoid adding the params to url
```

File: src/libs/Navigation/AppNavigator/ReportScreenWrapper.js
Comment (mountiny) [inline src/libs/Navigation/AppNavigator/ReportScreenWrapper.js:81]: This also needs to be functional component

File: src/libs/Navigation/AppNavigator/createResponsiveStackNavigator/ThreePaneView.js
Comment (mountiny) [inline src/libs/Navigation/AppNavigator/createResponsiveStackNavigator/ThreePaneView.js:25]: We prefer using `function`

File: src/libs/Navigation/AppNavigator/createResponsiveStackNavigator/CustomRouter.js
Comment (mountiny) [inline src/libs/Navigation/AppNavigator/createResponsiveStackNavigator/CustomRouter.js:23]: we prefer using `function` for the functional components

File: src/libs/Navigation/FreezeWrapper.js
Comment (mountiny) [inline src/libs/Navigation/FreezeWrapper.js:49]: displayName missing

**Review by mountiny — APPROVED** [review]
Couple of notes we should address in a follow up PRs

**Comment by WoLewicki** [comment]
Some other comments on what to look at in following PRs:
- check if patch for `@react-navigation/native` is still needed
- fix tests by either another approach for rendering heavy `ReportScreen` or properly handling navigation in tests
- look deeper at logic for `getIsReportFullyVisible` since part of it was needed due to drawer being able to cover the report screen
- fix flows for `Task` in RHP
- Remove `close` props from `HeaderWithBackButton` if it is meant not to be used there anymore
- check some weird navigation flows with browser back buttons and RHP

File: src/ROUTES.js
Comment (WoLewicki) [reply src/ROUTES.js:67]: cc @adamgrzybowski since you added it iirc.

**Comment by kok-hong** [comment]
.

**Comment by kok-hong** [comment]
staging.new.expensify.com/settings

**Comment by kok-hong** [comment]
[staging.new.expensify.com/settings](staging.new.expensify.com/settings)

File: src/libs/Navigation/FreezeWrapper.js
Comment (roryabraham) [inline src/libs/Navigation/FreezeWrapper.js:6]: Also, why is `withWindowDimensions` used here? I don't see any uses of the props. Is it just so it re-renders on resize? Or could we remove it?

File: src/components/FlatList/index.android.js
Comment (amyevans) [reply src/components/FlatList/index.android.js:33]: Hmm this is a WIP review I had started a few days ago that I tried to cancel just now because I never got around to finishing it. Sorry for the noise! (Not sure how to remove it now...)

File: src/libs/Navigation/FreezeWrapper.js
Comment (adamgrzybowski) [reply src/libs/Navigation/FreezeWrapper.js:6]: Good catch! I think that may be some leftovers. I am going to remove it and test if everything is working correctly, but it should be okay

File: src/ROUTES.js
Comment (adamgrzybowski) [reply src/ROUTES.js:67]: Before the refactor the `/r` and `/r/:id` routes were handled by different screens in the linking config. Now both routes are handled by the `ReportScreenWrapper` that accepts both cases, with or without id, and sets up id if necessary.

File: src/components/DragAndDrop/index.js
Comment (kosmydel) [reply src/components/DragAndDrop/index.js:74]: Hey, I verified it. In current logic, it is possible to call `addEventListeners()` twice, resulting in possibly doubling adding/removing the listeners.

Currently, registering a listener with a named function doesn't lead to the duplication of the listeners ([source1](https://www.js-craft.io/blog/javascript-addeventlistener-will-not-duplicate-named-functions/), [some context](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#memory_issues)). I tested it locally, and it worked as expected (Chrome 114):

<details>
  <summary>Testing results</summary>
  
**Plan**

1. Executing several (5) times `addEventListeners()` in the `componentDidMount()`.

2. Added two different listeners, one named, one arrow:
  ```javascript
  document.addEventListener('dragover', this.dropZoneDragListener);
  document.addEventListener('dragover', () => {
      console.log('dropZoneDragListener with arrow function');
  });
  ```

**Results:**
Screenshot 2023-07-12 at 12 47 02
The arrow function was called several times per event, and the named function only once.
  
</details>

Here is a possible solution, to eliminate doubling adding listeners.

<details>
  <summary>Possible solution</summary>
  
**Plan**

I can easily eliminate the possibility of calling the `this.addEventListeners` twice by updating the definition of `componentDidUpdate`.
```javascript
  componentDidUpdate(prevProps) {
      const isDisabled = this.props.disabled;
      if (this.props.isFocused === prevProps.isFocused && isDisabled === prevProps.disabled) {
          return;
      }

      if (!this.props.isFocused || isDisabled) {
          this.removeEventListeners();
      } else {
          this.addEventListeners();
      }
  }
```
  
In this solution, we might call the `this.removeEventListeners` twice anyway.

</details>

Please confirm, if we want to change it or should I look for a different solution.

Comment (mountiny) [reply src/components/DragAndDrop/index.js:74]: Can you please bring this to Slack as this might be easily lost here, thanks!

Comment (roryabraham) [reply src/components/DragAndDrop/index.js:74]: Agreed, @kosmydel please bring this up in slack following the *Proposal/Problem/Solution* format. Thanks! 🙇🏼

Comment (kosmydel) [reply src/components/DragAndDrop/index.js:74]: [Done](https://expensify.slack.com/archives/C01GTK53T8Q/p1689234806528439)

File: src/libs/Navigation/AppNavigator/AuthScreens.js
Comment (s77rt) [inline src/libs/Navigation/AppNavigator/AuthScreens.js:187]: This caused a regression https://github.com/Expensify/App/issues/20709. We should have kept using `getNavigationModalCardStyle`. We modified this function in https://github.com/Expensify/App/pull/19682 but that change was not caught after we merged `main` https://github.com/Expensify/App/pull/19263/commits/24d3f342aae426725bd6f33b4c9fab87381e29bb#diff-5b711055e18a2a6ec042cebe121391e826554644d2965d3365250d90d5d39c32

**Pull request merged**
